lib/Sema/BuilderTransform.cpp

```diff
@@ -459,6 +459,33 @@ class BuilderClosureVisitor
                                  /*isTopLevel=*/true);
   }
 
+  /// If there is a #available in the condition, the 'then' will be
+  /// wrapped in a call to buildLimitedAvailability(_:), if available.
+  /// If this is instead a #unavailability condition, the 'else' will
+  /// be wrapped instead.
+  Expr *wrapInLimitedAvailabilityIfNeeded(IfStmt *ifStmt, Expr *arg, 
+                                          bool isUnavailability = false) {
+    auto availabilityCond = findAvailabilityCondition(ifStmt->getCond());
+    bool supportsAvailable = builderSupports(ctx.Id_buildLimitedAvailability);
+
+    auto apiAvailability = availabilityCond->getAvailability();
+    if (availabilityCond && 
+        apiAvailability->isUnavailability() == isUnavailability &&
+        supportsAvailable) {
+      SourceLoc endLoc;
+      if (isUnavailability) {
+        endLoc = ifStmt->getEndLoc();
+      } else {
+        endLoc = ifStmt->getThenStmt()->getEndLoc();
+      }
+      return buildCallIfWanted(
+          ifStmt->getThenStmt()->getEndLoc(), ctx.Id_buildLimitedAvailability,
+          { arg }, { Identifier() });
+    } else {
+      return arg;
+    }
```

Comment:
Style nit: LLVM style recommends using early returns/exits, so you could `return arg;` after a check and unindent the other logic.